script.js

```diff
@@ -1,4 +1,262 @@
+      // value and rank of card is the same for cards 2 to 10.
+      SumOfCards = SumOfCards + currentCard.rank;
+    } else if (currentCard.rank >= 11 && currentCard.rank <= 13) {
+      // value of J,Q,K is 10.
+      SumOfCards = SumOfCards + 10;
+    } else if (currentCard.rank == 1) {
+      acesInHand = acesInHand + 1;
+      SumOfCards = SumOfCards + 11;
+    }
+    index = index + 1;
+  }
+
+  if (SumOfCards > maxSumOfCards && acesInHand > 0) {
+    var index = 0;
+    while (index < acesInHand) {
+      SumOfCards = SumOfCards - 10;
+      if (SumOfCards <= maxSumOfCards) {
+        break;
+      }
+      index = index + 1;
+    }
+  }
+
+  return SumOfCards;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (endGame == true) {
+    endGame = true;
+    console.log(endGame);
+    return "Please refresh to play the game.";
+  }
+
+  // if players have no cards, deal a card
+  if (playerHand.length == 0) {
+    playerHand = dealCardToHand(playerHand);
+    dealerHand = dealCardToHand(dealerHand);
+    return `You got the ${playerHand[0].name} of ${playerHand[0].suit} and the dealer got the ${dealerHand[0].name} of ${dealerHand[0].suit} <br> Please click Submit again to deal another card.`;
+  } else if (playerHand.length == 1) {
+    playerHand = dealCardToHand(playerHand);
+    dealerHand = dealCardToHand(dealerHand);
+    return `You got the ${playerHand[0].name} of ${playerHand[0].suit} and the ${playerHand[1].name} of ${playerHand[1].suit}. The dealer got the ${dealerHand[0].name} of ${dealerHand[0].suit} and another card.<br>Please input 'hit' or 'stand' as your choice.`;
+  }
+
+  if (getSumOfHand(playerHand) == maxSumOfCards) {
+    endGame = true;
+    return "Player wins game by having a total of 21.";
+  }
+  if (getSumOfHand(dealerHand) == maxSumOfCards) {
+    endGame = true;
+    return "Dealer wins game by having a total of 21.";
+  }
+
+  // if (playerChoseStand == true) {
+  //   playerChoseStand = true;
+  //   if (input != "hit" && input != "stand") {
+  //     return "Please input either 'hit' or 'stand' to continue.";
+  //   }
+
+  if (input == "hit") {
+    playerHand = dealCardToHand(playerHand);
+    index2 += 1;
+    if (getSumOfHand(playerHand) > maxSumOfCards) {
+      return `You got the ${playerHand[0].name} of ${playerHand[0].suit} and the ${playerHand[1].name} of ${playerHand[1].suit} ${others}. The dealer got the ${dealerHand[0].name} of ${dealerHand[0].suit} and another card.<br> Dealer wins! You have a sum of more than 21 and lost. Please try again.`;
+    } else {
+      others += `and the ${playerHand[index2].name} of ${playerHand[index2].suit}`;
+      return `You got the ${playerHand[0].name} of ${playerHand[0].suit} and the ${playerHand[1].name} of ${playerHand[1].suit} ${others}. The dealer got the ${dealerHand[0].name} of ${dealerHand[0].suit} and another card<br>Please input 'hit' or 'stand' as your choice.`;
```

Comment:
Instead of having many lines of code for the return statement, you could create a helper function that generates custom messages depending on the gameMode. e.g.
generateMessage(gameMode)